### ISSUE TITLE
[BUG] Activation fails if there are no changes on Check MK server that have to be activated

### ISSUE THREAD

**Issue by msekania** [Checkmk/ansible-collection-tribe29.checkmk#312]
**Describe the bug**

 Activation fails if there are no changes on Check MK server that have to be activated

**Component Name**

 `activation`

**Ansible Version**

```console
$ ansible --version
ansible [core 2.14.1]
```

**Checkmk Version**

```console
2.1.11p (CRE)
```

**Collection Version**

```console
$ ansible-galaxy collection list

Collection           Version
-------------------- -------
ansible.posix        1.4.0  
ansible.utils        2.9.0  
community.crypto     2.10.0 
community.general    5.7.0  
community.mysql      3.5.1  
tribe29.checkmk      0.20.0 

```
**Environment**

**To Reproduce**

just run activation when there are no changes to be activated

test_activate.yml:
```yaml
- name: Test Activation
  hosts: localhost
  gather_facts: false

  tasks:
    - name: Activate check_mk changes
      tribe29.checkmk.activation:
        server_url: "https://{{ my_server }}/"
        site: "{{ my_site }}"
        automation_user: "{{ automation_user }}"
        automation_secret: "{{ automation_secret }}"
        force_foreign_changes: 'true'
        sites:
          - "{{ site }}"
        validate_certs: false
      run_once: true
      delegate_to: localhost

```

`ansible-playbook test_activate.yml -vvv`

**Expected behavior**

should pass without a problem

**Actual behavior**

```console
Using module file XXXXXXXXX/collections/ansible_collections/tribe29/checkmk/plugins/modules/activation.py
Pipelining is enabled.
<localhost> ESTABLISH LOCAL CONNECTION FOR USER: XXXXXXXX
<localhost> EXEC /bin/sh -c '/usr/bin/python3 && sleep 0'
The full traceback is:
Traceback (most recent call last):
  File "<stdin>", line 107, in <module>
  File "<stdin>", line 99, in _ansiballz_main
  File "<stdin>", line 47, in invoke_module
  File "/usr/lib/python3.10/runpy.py", line 224, in run_module
    return _run_module_code(code, init_globals, run_name, mod_spec)
  File "/usr/lib/python3.10/runpy.py", line 96, in _run_module_code
    _run_code(code, mod_globals, init_globals,
  File "/usr/lib/python3.10/runpy.py", line 86, in _run_code
    exec(code, run_globals)
  File "/tmp/ansible_tribe29.checkmk.activation_payload_n6b6tf1q/ansible_tribe29.checkmk.activation_payload.zip/ansible_collections/tribe29/checkmk/plugins/modules/activation.py", line 150, in <module>
  File "/tmp/ansible_tribe29.checkmk.activation_payload_n6b6tf1q/ansible_tribe29.checkmk.activation_payload.zip/ansible_collections/tribe29/checkmk/plugins/modules/activation.py", line 146, in main
  File "/tmp/ansible_tribe29.checkmk.activation_payload_n6b6tf1q/ansible_tribe29.checkmk.activation_payload.zip/ansible_collections/tribe29/checkmk/plugins/modules/activation.py", line 138, in run_module
  File "/tmp/ansible_tribe29.checkmk.activation_payload_n6b6tf1q/ansible_tribe29.checkmk.activation_payload.zip/ansible_collections/tribe29/checkmk/plugins/modules/activation.py", line 116, in post
  File "/tmp/ansible_tribe29.checkmk.activation_payload_n6b6tf1q/ansible_tribe29.checkmk.activation_payload.zip/ansible_collections/tribe29/checkmk/plugins/module_utils/api.py", line 62, in _fetch
  File "/usr/lib/python3.10/json/__init__.py", line 346, in loads
    return _default_decoder.decode(s)
  File "/usr/lib/python3.10/json/decoder.py", line 337, in decode
    obj, end = self.raw_decode(s, idx=_w(s, 0).end())
  File "/usr/lib/python3.10/json/decoder.py", line 355, in raw_decode
    raise JSONDecodeError("Expecting value", s, err.value) from None
json.decoder.JSONDecodeError: Expecting value: line 1 column 1 (char 0)
fatal: [localhost]: FAILED! => {
    "changed": false,
    "module_stderr": "Traceback (most recent call last):\n  File \"<stdin>\", line 107, in <module>\n  File \"<stdin>\", line 99, in _ansiballz_main\n  File \"<stdin>\", line 47, in invoke_module\n  File \"/usr/lib/python3.10/runpy.py\", line 224, in run_module\n    return _run_module_code(code, init_globals, run_name, mod_spec)\n  File \"/usr/lib/python3.10/runpy.py\", line 96, in _run_module_code\n    _run_code(code, mod_globals, init_globals,\n  File \"/usr/lib/python3.10/runpy.py\", line 86, in _run_code\n    exec(code, run_globals)\n  File \"/tmp/ansible_tribe29.checkmk.activation_payload_n6b6tf1q/ansible_tribe29.checkmk.activation_payload.zip/ansible_collections/tribe29/checkmk/plugins/modules/activation.py\", line 150, in <module>\n  File \"/tmp/ansible_tribe29.checkmk.activation_payload_n6b6tf1q/ansible_tribe29.checkmk.activation_payload.zip/ansible_collections/tribe29/checkmk/plugins/modules/activation.py\", line 146, in main\n  File \"/tmp/ansible_tribe29.checkmk.activation_payload_n6b6tf1q/ansible_tribe29.checkmk.activation_payload.zip/ansible_collections/tribe29/checkmk/plugins/modules/activation.py\", line 138, in run_module\n  File \"/tmp/ansible_tribe29.checkmk.activation_payload_n6b6tf1q/ansible_tribe29.checkmk.activation_payload.zip/ansible_collections/tribe29/checkmk/plugins/modules/activation.py\", line 116, in post\n  File \"/tmp/ansible_tribe29.checkmk.activation_payload_n6b6tf1q/ansible_tribe29.checkmk.activation_payload.zip/ansible_collections/tribe29/checkmk/plugins/module_utils/api.py\", line 62, in _fetch\n  File \"/usr/lib/python3.10/json/__init__.py\", line 346, in loads\n    return _default_decoder.decode(s)\n  File \"/usr/lib/python3.10/json/decoder.py\", line 337, in decode\n    obj, end = self.raw_decode(s, idx=_w(s, 0).end())\n  File \"/usr/lib/python3.10/json/decoder.py\", line 355, in raw_decode\n    raise JSONDecodeError(\"Expecting value\", s, err.value) from None\njson.decoder.JSONDecodeError: Expecting value: line 1 column 1 (char 0)\n",
    "module_stdout": "",
    "msg": "MODULE FAILURE\nSee stdout/stderr for the exact error",
    "rc": 1
}
```

**Screenshots**

**Additional context**

**Comment by msekania** [comment]
I have also check with REST API

Response body is:
```json
{
  "title": "The operation has failed.",
  "status": 422,
  "detail": "Currently there are no changes to activate."
}
```

In essence this should not be the error for Ansible!

**Comment by robin-checkmk** [comment]
I can confirm this issue. There seems to be a return code missing in the answer handling.
There is currently a [PR](https://github.com/tribe29/ansible-collection-tribe29.checkmk/pull/298) queued, that touches this module. I would want to wait for that merge, before building the fix.

**Comment by msekania** [comment]
@robin-checkmk,

I have checked and code 422 is already included in #298

**Comment by robin-checkmk** [comment]
This should be fixed in `0.22.0`.
